src/storage.js

```diff
@@ -0,0 +1,432 @@
+/*
+ * This file is part of Privacy Badger <https://www.eff.org/privacybadger>
+ * Copyright (C) 2014 Electronic Frontier Foundation
+ *
+ * Privacy Badger is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * Privacy Badger is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Privacy Badger.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+require.scopes.storage = (function() {
+
+
+/**
+ * # Storage Objects
+ *
+ * snitch_map is our collection of potential tracking base_domains.
+ * The key is a base domain (ETLD+1) and the value is an array of first
+ * party domains on which this tracker has been seen.
+ * it looks like this:
+ * { 
+ *   "third-party.com": ["a.com", "b.com", "c.com"], 
+ *   "eviltracker.net": ["eff.org", "a.com"]
+ * }
+ *
+ * action_map is where we store the action for each domain that we have 
+ * decided on an action for. Each subdomain gets its own entry. For example:
+ * {
+ *   "google.com": { heuristicAction: "block", dnt: false, userAction: ""}
+ *   "fonts.google.com": { heuristicAction: "cookieblock", dnt: false, userAction: ""}
+ *   "apis.google.com": { heuristicAction: "cookieblock", dnt: false, userAction: "user_block"}
+ *   "widget.eff.org": { heuristicAction: "block", dnt: true, userAction: ""}
+ * }
+ *
+ * cookieblock_list is where we store the current cookie block list as 
+ * downloaded from eff.org. The keys are the domains which should be blocked.
+ * The values are simply 'true'
+ *
+ * { 
+ *   "maps.google.com": true,
+ *   "creativecommons.org": true,
+ * }
+ **/
+
+// Cache of BadgerStorage objects
+var badgerPen = {};
+
+
+var initialize = function(){
+  console.log('loading badgers into the pen');
+  var storage_objects = [
+    "snitch_map",
+    "action_map",
+    "cookieblock_list",
+    "supercookie_domains",
+    "dnt_hashes",
+    "settings_map"
+  ];
+
+  for(var i = 0; i < storage_objects.length; i++){
+    _initializeCache(storage_objects[i]);
+  }
+};
+
+var getScore = function(action){
+  switch(action){
+    case pb.NO_TRACKING: return 0;
+    case pb.ALLOW: return 1;
+    case pb.BLOCK: return 2;
+    case pb.COOKIEBLOCK: return 3;
+    case pb.DNT: return 4;
+    default: return 5; 
```

Comment:
yea, exactly. My assumption is that we will never get a non existent action through here. Though it might be safer to rewrite this to not make that assumption.